rooms.js

```diff
@@ -439,6 +439,36 @@ class GlobalRoom {
 		}
 		return this.formatList;
 	}
+	get configRankList() {
+		if (Config.nocustomgrouplist) return '';
+
+		// putting the resultant object in Config would enable this to be run again should config.js be reloaded.
+		if (Config.rankList) {
+			return Config.rankList;
+		}
+		let rankList = [];
+
+		for (let rank in Config.groups) {
+			if (!Config.groups[rank] || !rank) continue;
+
+			let tarGroup = Config.groups[rank];
+			let groupType = tarGroup.addhtml || (!tarGroup.mute && !tarGroup.root) ? 'normal' : (tarGroup.root || tarGroup.declare) ? 'leadership' : 'staff';
+
+			rankList.push({symbol: rank, name: (Config.groups[rank].name || null), type: groupType}); // send the first character in the rank, incase they put a string several characters long
+		}
+
```

Comment:
_glomps @DeathlyPlays _
OMG it's you!!! the famous youtuber!! https://www.youtube.com/watch?v=IIlFwamCUcA
_hi!.. *squeek!* cant breathe ^-^ _
i-i am such a big fan!! you're likeee my favorite kid to watch on youtube!! 
i w-watch your videos like everyday and pull out my little chinchin 😊